### ISSUE TITLE
ViewModelInejectors

### ISSUE THREAD

**Issue by bobpf** [onepointsixtwo/dagger_viewmodel#2]
Tried following your example in the README but the ViewModelInejectors was never generated.  Also, it would be nice to see a more robust example of how you would use that guy in Dagger2.  Thanks.

**Comment by onepointsixtwo** [comment]
Heya. I'm sorry if the instructions aren't too clear. I'll try and improve them when I get a chance. 

I threw together an example project with the minimal setup required to use the library here:
https://github.com/onepointsixtwo/dagger_viewmodel_example

It gives a minimal dagger setup alongside the code to inject a ViewModel subclass with a dependency injected into it, and also shows it works by using a value from the dependency within the main activity.

I'm afraid whilst I was creating the project I discovered two unfortunate things, and I'm a little busy doing an MSc at the moment, so I don't have time to immediately fix either (though it would be possible to fork the library and make these changes). These are:

1) It only supports injection into an actual Java field and not a setter. This is a problem in that it will not work correctly with Kotlin. Kotlin 'fields' are actually auto-generated getters and setters for a hidden underlying field not visible from Java, and this creates problems, so if you're trying to inject into a Kotlin fragment / activity, it'll fail with an exception of something like "only injection into fields is supported".

2) It does not support AndroidX. I wrote this library before AndroidX was a thing, and its dependent on the older version of android.arch from before AndroidX, so attempting to use it with will give a compilation error. 

I'll try and address both of those issues when I can, but I'm not entirely sure when that will be. Apologies if either are problems for what you're attempting to achieve.
